libraries/AP_BattMonitor/AP_BattMonitor_Generator.cpp

```diff
@@ -0,0 +1,38 @@
+#include <AP_HAL/AP_HAL.h>
+#include <AP_Common/AP_Common.h>
+#include <AP_Math/AP_Math.h>
+#include "AP_BattMonitor.h"
+#include "AP_BattMonitor_Generator.h"
+
+extern const AP_HAL::HAL& hal;
+
+// read - read the voltage and current
+void AP_BattMonitor_Generator::read()
+{
+    _state.healthy = false;
+
+#if RICHENPOWER_ENABLED
+    AP_RichenPower *richenpower = AP::richenpower();
```

Comment:
I probably would have called this variable generator instead of richenpower.  Not a big deal though.

Reply:
My reasoning was that if another generator was added we didn't want naming conflicts.

Reply:
I think if more generators are added we should do a front-end/back-end split in the AP_Generator library.